app/records/mixins.py

```diff
@@ -0,0 +1,293 @@
+                f"/pages/{settings.WAGTAIL_HOME_PAGE_ID}"
+            )
+        except Exception as e:
+            logger.error(f"Failed to fetch global alerts: {e}")
+            return {}
+
+    def get_context_data(self, **kwargs):
+        """Add global alerts to context."""
+        context = super().get_context_data(**kwargs)
+        context["global_alert"] = self.get_global_alerts()
+        return context
+
+
+class SubjectsEnrichmentMixin:
+    """Mixin for enriching records with subjects data."""
+
+    def enrich_record_subjects(self, record: Record) -> None:
+        """
+        Enrich a record with subjects data in-place.
+
+        Fetches related articles and content for the record's subject tags
+        from the Wagtail CMS and attaches them to the record object.
+
+        Args:
+            record: The record to enrich (modified in-place)
+        """
+        if record.subjects:
+            subjects_enrichment = get_subjects_enrichment(
+                record.subjects, limit=settings.MAX_SUBJECTS_PER_RECORD
+            )
+            record._subjects_enrichment = subjects_enrichment
+        else:
+            record._subjects_enrichment = {}
+
+    def get_context_data(self, **kwargs):
+        """Enrich record with subjects before adding to context."""
+        context = super().get_context_data(**kwargs)
+        if "record" in context:
+            self.enrich_record_subjects(context["record"])
+
+        return context
+
+
+class RelatedRecordsMixin:
+    """Mixin for adding related records to context."""
+
+    related_records_limit = 3
+
+    def get_related_records(self, record: Record) -> list[Record]:
+        """
+        Fetch related records using subjects first, then series as fallback.
+
+        Attempts to find related records by shared subjects. If fewer than the
+        requested limit are found, backfills the remaining slots with records
+        from the same archival series.
+
+        Args:
+            record: The record to find relations for
+
+        Returns:
+            List of related Record objects (up to limit)
+        """
+        related_records = get_related_records_by_subjects(
+            record, limit=self.related_records_limit
+        )
+
+        # Backfill from series if needed
+        if len(related_records) < self.related_records_limit:
+            remaining_slots = self.related_records_limit - len(related_records)
+            series_records = get_related_records_by_series(
+                record, limit=remaining_slots
+            )
+            related_records.extend(series_records)
+
+        return related_records
+
+    def get_context_data(self, **kwargs):
+        """Add related records to context."""
+        context = super().get_context_data(**kwargs)
+        if "record" in context:
+            context["related_records"] = self.get_related_records(
+                context["record"]
+            )
+        return context
```

Comment:
There might be a bit too much mixin layering for RecordDetailView(), which could make debugging tricky later on. Would it be simpler and cleaner to just call add_related_records_context(context, record) directly instead - and move the functionality to `app/records/related.py` ?